### DISCUSSION TITLE
Can i make  custom layout update programmatically that is based on a certain page Layout?

### DISCUSSION THREAD

**Discussion by csl-schlewitz** [OpenMage/magento-lts#4014] [Questions & Answers]
After i setup a new page layout in my modules config.xml

```xml
<page>
            <layouts>
                <category_one_column_grid module="tenretni_catalog" translate="label">
                    <label>Category (full grid)</label>
                    <template>page/1column-category.phtml</template>
                    <layout_handle>category_one_column</layout_handle>
                </category_one_column_grid>
            </layouts>
</page>
```
i can assign layout updates for a category on the admin page...

![Screenshot 2024-05-27 161426](https://github.com/OpenMage/magento-lts/assets/73691764/3f0e0771-0ab4-4fb7-bb01-035624556544)

This generates a lot of copy- paste workload, so i was wondering if i can make something like  that within the layout xml's

i found soemthing similar in the paypal.xml

where  set of layout adjustments
```xml
<SHORTCUT_popup>
        <reference name="product.tooltip">
            <block type="page/html_wrapper" name="product.info.addtocart.paypal.wrapper" translate="label">
                <block type="paypal/express_shortcut" name="product.info.addtocart.paypal" template="paypal/express/shortcut.phtml">
                    <action method="setIsInCatalogProduct"><value>1</value></action>
                    <action method="setShowOrPosition"><value>after</value></action>
                </block>
            </block>
        </reference>
    </SHORTCUT_popup>
```
are invoke ontop of another page-block
```xml
<checkout_onepage_success>
        <update handle="SHORTCUT_popup" />
    </checkout_onepage_success>
```

But so far non of the update handles  i found in the code references  a page-layout like
page_one_column, page_two_columns_left

I know that the page-block for the category  page is either
**catalog_category_default** or **catalog_category_layered**

i also tried adding a node to the page.xml
```xml
<category_one_column translate="label">
    <reference name="root">
        <action method="setTemplate">
            <template>page/1column-category.phtml</template>
        </action>
        <!-- Mark root page block that template is applied -->
        <action method="setIsHandle">
            <applied>1</applied>
        </action>
        
        <reference name="catalog_category_default">
            <reference name="product_list">
                <action method="setDefaultDirection"><dir>asc</dir></action>
                <action method="setTemplate">
                    <template>catalog/product/grid.phtml</template>
                </action>
                <reference name="category.list.product">
                    <action method="setTemplate">
                        <template>catalog/product/view/card/small.phtml</template>
                    </action>
                </reference>
            </reference>
        </reference>
    </reference>
   </category_one_column>
```
To  reference the page block  and invoke my adjustment
But none of the changes seem to have any effect.

the page definiton in this place seem utterly useless, aswell as the standard pages,... changes or removing those dont seem to have any effect,...
(since the page-layout, template and handle in itself is defined in the config.xml)

does anyone have any insight about that or is there no other way then to add custom layout updated within the category mangement,...?

**Comment by kiatng** [comment]
I have never programmatically update a catalog layout. If I were to start, I would look at how it's done in  https://github.com/OpenMage/magento-lts/blob/1f32c70e0e791511b89f217a51c014e625513658/app/code/core/Mage/Catalog/controllers/CategoryController.php#L117-L171

Line 124: `if ($settings->getCustomDesign()) {`, I didn't look too far, but may be you can make use of `$settings->setCustomDesign($myDesign)`.

Unrelated: see [syntax highlight in markdown](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting)
